scipy/stats/_binomtest.py

```diff
@@ -313,16 +312,59 @@ def binomtest(k, n, p=0.5, alternative='two-sided'):
             # special case as shortcut, would also be handled by `else` below
             pval = 1.
         elif k < p * n:
-            i = np.arange(np.ceil(p * n), n+1)
-            y = np.sum(binom.pmf(i, n, p) <= d*rerr, axis=0)
+            y = n-_binary_search_for_binom_tst(lambda x1: -binom.pmf(x1, n, p),
+                                               -d*rerr, np.ceil(p * n), n)
             pval = binom.cdf(k, n, p) + binom.sf(n - y, n, p)
         else:
-            i = np.arange(np.floor(p*n) + 1)
-            y = np.sum(binom.pmf(i, n, p) <= d*rerr, axis=0)
+            y = _binary_search_for_binom_tst(lambda x1: binom.pmf(x1, n, p),
+                                             d*rerr, 0, np.floor(p * n) + 1,
+                                             True)+1
             pval = binom.cdf(y-1, n, p) + binom.sf(k-1, n, p)
 
         pval = min(1.0, pval)
 
     result = BinomTestResult(k=k, n=n, alternative=alternative,
                              proportion_estimate=k/n, pvalue=pval)
     return result
+
+
+def _binary_search_for_binom_tst(a, d, lo, hi, asc_order=False):
+    r"""
+    Conducts an implicit binary search on a funcion specified by `a.
+
+    Meant to be used on the binomial PMF for the case of two-sided tests
+    to obtain the value on the other side of the mode where the tail prbability
```

Comment:
```suggestion
    to obtain the value on the other side of the mode where the tail probability
```

I'm a bit confused on the inclusion of `on the other side of the mode` part, maybe I missed this part but isn't the calculation being done on the ***same*** side of the mode?

Reply:
No, for the two sided test, we want to find the point on the other side that is as extreme as the one we have on the current side. Hence, its the value as extreme as observation on other side of the binomial mode.